internal/landing/landing.go

```diff
@@ -8,22 +8,69 @@ package landing
 
 import (
 	"embed"
+	"fmt"
+	"html/template"
 	"io/fs"
 	"net/http"
+	"net/url"
+
+	"github.com/talos-systems/discovery-service/internal/state"
 )
 
 //go:embed "html/index.html"
 var static embed.FS
 
+//go:embed "templates/inspect.html.tmpl"
+var inspectTemplate []byte
+
+// ClusterInspectData represents all affiliate data asssociated with a cluster.
+type ClusterInspectData struct {
+	ClusterID  string
+	Affiliates []*state.AffiliateExport
+}
+
+// InspectHandler provides data to inspect a cluster.
+func InspectHandler(w http.ResponseWriter, req *http.Request, state *state.State) error {
+	clusterData := ClusterInspectData{}
+
+	if formData, err := url.ParseQuery(req.URL.RawQuery); err == nil {
+		clusterData.ClusterID = formData.Get("clusterID")
+		fetchedCluster := state.GetCluster(clusterData.ClusterID)
+		clusterData.Affiliates = fetchedCluster.List()
+	} else {
+		return err
+	}
+
+	inspectPage, err := template.New("inspect").Parse(string(inspectTemplate))
+	if err != nil {
+		return err
+	}
+
+	if err := inspectPage.Execute(w, clusterData); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // Handler returns static landing page handler.
-func Handler() http.Handler {
+func Handler(state *state.State) http.Handler {
 	subfs, err := fs.Sub(static, "html")
 	if err != nil {
 		panic(err)
 	}
 
 	mux := http.NewServeMux()
+
 	mux.Handle("/", http.FileServer(http.FS(subfs)))
 
+	// wrapper to pass state
+	mux.HandleFunc("/inspect", func(w http.ResponseWriter, r *http.Request) {
+		if err := InspectHandler(w, r, state); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "%s", err)
```

Comment:
Do we want to return that error to the user, or we want to show some generics "oops" message and log internally instead?

Reply:
The error/reason might help the user make a proper request, this service will mostly be used by a technical person I think. Generic message also works fine

Reply:
@smira what do you think is the best way to go ahead?